### PR TITLE
Initial parser implementation

### PR DESCRIPTION
**Pull request by certik**
The parser and AST was taken from LPython. In subsequent commits we will start modifying it to parse C++.

### CODE REVIEW

**Comment by certik** [comment]
TODO: get `--show-ast` working.

**Comment by certik** [comment]
@czgdp1807, @Shaikh-Ubaid I don't have time to completely finish it. Go ahead and take it from here. We need to get `--show-ast` working and then start modifying the parser to parse C++ instead of Python. I would not bother with AST->ASR until we can parse enough of AST. So it's an orthogonal feature for now.

File: src/libasr/asdl_cpp.py [hunk 1]
```diff
@@ -2749,6 +2749,10 @@ def main(argv):
     elif subs["MOD"] == "AST":
          subs["MOD"] = "LFortran::AST"
          subs["lcompiler"] = "lfortran"
+    elif subs["MOD"] == "LC":
+        subs["MOD"] = "LC::AST"
+        subs["mod"] = "ast"
+        subs["lcompiler"] = "lpython"
```
Comment (Shaikh-Ubaid) [inline src/libasr/asdl_cpp.py:2755]: I think this should be `subs["lcompiler"] = "lc"`.

File: src/bin/lc.cpp [hunk 2]
```diff
@@ -713,7 +713,7 @@ int mainApp(int argc, const char **argv) {
     app.add_flag("--ast-print", ast_print, "Build ASTs and then pretty-print them");
     app.add_flag("--ast-dump-filter", ast_dump_filter, "Use with -ast-dump or -ast-print to dump/print"
      " only AST declaration nodes having a certain substring in a qualified name.");
-    app.add_flag("--show-ast", show_ast, "Show AST for the given file and exit");
+    app.add_flag("--show-clang-ast", show_clang_ast, "Show AST for the given file and exit");
```
Comment (Shaikh-Ubaid) [inline src/bin/lc.cpp:716]:
```suggestion
    app.add_flag("--show-clang-ast", show_clang_ast, "Show Clang AST for the given file and exit");
```

**Review by Shaikh-Ubaid — APPROVED** [review]
It looks good to me. Thank you so much for this. We can iterate on this.

**Pull request merged**